### ISSUE TITLE
Need ability to add a jumphost to SSH parameters to install RKE/RKE2/K3s into private networks

### ISSUE THREAD

**Issue by alexarnoldy** [harvester/harvester#1750]
**Is your feature request related to a problem? Please describe.**
My environment has mostly VLANs fronted with NAT routers, thus Harvester can't SSH to the nodes it deploys.

**Describe the solution you'd like**
I would like the ability to identify a jumphost (perhaps in the node template) that is accessible to Harvester as well as the nodes inside the private network. Then, Harvester could SSH to the nodes (i.e. ssh -J <user@jumphost> <user@target_node>) to install RKE/RKE2/K3s. It seems like Harvester would need to provide its public key so the user could put it into the authorized_keys file of the jumphost.

**Describe alternatives you've considered**
I think a cleaner way of handling this would be through cloud-init. Rancher could create the cluster, including a specific join token, then pass the needed info to Harvester. Harvester could inject the info plus first Master server IP as a command into the cloud-init config for the nodes.

**Additional context**
None.

**Comment by yasker** [comment]
@alexarnoldy I am not sure if I get the issue right. Are you talking about Rancher cannot access the VMs created by Harvester? In that case, RKE2 doesn't require ingress traffic to work, since it relies on cloud-init to bring up the Rancher agent on the VMs.

**Comment by alexarnoldy** [comment]
Thanks for the response on this. Let me test this out again to see if the issue has changed.

Previously, while deploying an RKE/RKE2/K3s cluster (sorry, I don't remember which it was but I can test it if needed) on Harvester, but on a VLAN that is behind a NAT firewall, the VMs deploy, but then the process stops progressing (from the Rancher UI) with a message that says it's waiting for SSH to become available (again, sorry, I don't remember the exact message).

Best regards,

--Alex.


Alex Arnoldy

Software Defined Infrastructure Solution Architect

[cid:c6e5423e-7c4e-43aa-9e63-5205f0ba06c5]

SUSE: Simplify, Modernize, Accelerate

SUSE One Partners

The Power of Many. Together as One.

mobile: 408.828.9797

email: ***@***.******@***.***>

________________________________

**Comment by yasker** [comment]
@alexarnoldy In Rancher, RKE1 provisioning requests SSH access, but RKE2 doesn't. Though RKE2 is not GAed yet, it will be soon in the end of March Rancher release.

**Comment by alexarnoldy** [comment]
@yasker, I hadn't noodled it out far enough, but now that you mention it, I think I submitted this request before Rancher could provision RKE2/K3s clusters. Around the same time I switched to primarily using a routed network for Harvester nodes, so I no longer saw the issue. I guess the future of this request depends on the future of RKE vs. RKE2. Being able to edit the SSH command that would be used to provision the RKE nodes would allow specifying jumphosts, unique SSH keys, non-standard ports, and more. If RKE is expected to remain rather static while RKE2 will get more developer attention, this request might not offer enough value to the community. Feel free to keep or close it as you see fit.
